enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/InternalAPIKeyAuthenticator.java

```diff
@@ -203,8 +207,31 @@ public AuthenticationContext authenticate(RequestContext requestContext) throws
                     try {
                         api = validateAPISubscription(apiContext, apiVersion, payload, splitToken,
                                 false);
+                        if (api != null) {
+                            String context = requestContext.getMatchedAPI().getBasePath();
+                            String uuid = requestContext.getMatchedAPI().getUuid();
+                            String apiTenantDomain = FilterUtils.getTenantDomainFromRequestURL(context);
+                            SubscriptionDataStore datastore = SubscriptionDataHolder.getInstance()
+                                    .getTenantSubscriptionStore(apiTenantDomain);
+                            API subscriptionDataStoreAPI = datastore.getApiByContextAndVersion(uuid);
+                            if (subscriptionDataStoreAPI != null &&
+                                    APIConstants.LifecycleStatus.BLOCKED
+                                            .equals(subscriptionDataStoreAPI.getLcState())) {
+                                FilterUtils.setErrorToContext(requestContext,
+                                        GeneralErrorCodeConstants.API_BLOCKED_CODE,
+                                        APIConstants.StatusCodes.SERVICE_UNAVAILABLE.getCode(),
+                                        GeneralErrorCodeConstants.API_BLOCKED_MESSAGE,
+                                        GeneralErrorCodeConstants.API_BLOCKED_DESCRIPTION);
+                                throw new APISecurityException(APIConstants.StatusCodes.SERVICE_UNAVAILABLE.getCode(),
+                                        GeneralErrorCodeConstants.API_BLOCKED_CODE,
+                                        GeneralErrorCodeConstants.API_BLOCKED_MESSAGE);
+                            }
+                            log.debug("Internal Key Authentication is successful.");
+                        }
+                    } catch (APISecurityException e) {
+                        throw e;
                     } finally {
-                        log.debug("Internal Key authentication successful.");
+                        log.debug("Internal Key authentication complete.");
```

Comment:
Internal Key authentication complete. --> Internal Key Authentication is completed.

Reply:
```suggestion
                        log.debug("Internal Key authentication is completed.");
```